source/slang-capture-replay/output-stream.cpp

```diff
@@ -0,0 +1,52 @@
+#include "output-stream.h"
+#include "capture_utility.h"
+
+namespace SlangCapture
+{
+    FileOutputStream::FileOutputStream(const std::string& filename, bool append)
+    {
+        Slang::String path(filename.c_str());
+        Slang::FileMode fileMode = append ? Slang::FileMode::Append : Slang::FileMode::Create;
+        Slang::FileAccess fileAccess = Slang::FileAccess::Write;
+        Slang::FileShare fileShare = Slang::FileShare::None;
+
+        SlangResult res = m_fileStream.init(path, fileMode, fileAccess, fileShare);
+
+        if (res != SLANG_OK)
+        {
+            SlangCapture::slangCaptureLog(SlangCapture::LogLevel::Error, "Failed to open file %s\n", filename.c_str());
+            std::abort();
+        }
+    }
+
+    FileOutputStream::~FileOutputStream()
+    {
+        m_fileStream.close();
+    }
+
+    void FileOutputStream::write(const void* data, size_t len)
+    {
+        SLANG_CAPTURE_ASSERT(m_fileStream.write(data, len));
```

Comment:
Does `write` return zero when `len` is zero?

Reply:
Yes, it only return false when the stream cannot be written.

Reply:
Here is the implementation of "write()"
```
SlangResult FileStream::write(const void* buffer, size_t length)
{
    auto bytesWritten = fwrite(buffer, 1, length, m_handle);
    return (bytesWritten == length) ? SLANG_OK : SLANG_FAIL;
}
```
If the `len` is zero, it will return a value, zero, and it will trigger an assertion failure and it will crash the application unnecessarily.
I like to suggest to change to the following,
```
    void FileOutputStream::write(const void* data, size_t len)
    {
        if (len == 0) return;
        SLANG_CAPTURE_ASSERT(m_fileStream.write(data, len));
    }
```